### DISCUSSION TITLE
CVE-2023-4863

### DISCUSSION THREAD

**Discussion by crashr** [qutebrowser/qutebrowser#7915] [General] [answered]
Are we affected? If yes, would it be possible to make use of Chromiums image.webp.enabled to workaround this?

Edit: No, we can't since image.webp.enabled is in Firefox, not in Chromium. Is there a Chromium pendant?

**Comment by The-Compiler** [comment, marked as answer]
Most likely, yes. For context: CVE-2023-4863, [Chrome Releases: Stable Channel Update for Desktop](https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html).

QtWebEngine fixed it here:

[[Backport] CVE-2023-4863: Heap buffer overflow in WebP (I31c36dbf) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine-chromium/+/503192)

It was commited to 87-based, 108-based and 112-based, so I'd expect it to end up in [Qt 5.15.16](https://wiki.qt.io/Qt_5.15_Release#Qt_5.15_LTS_Commercial_Releases), [Qt 6.5.3](https://lists.qt-project.org/pipermail/releasing/2023-September/002995.html) and [Qt 6.6](https://wiki.qt.io/Qt_6.6_Release), with 6.5.3 and 6.6 scheduled to released somewhen later this month.

I've also asked Archlinux to backport the patch: [FS#79703 : [qt6-webengine] [security] (patch) CVE-2023-4863: libwebp: OOB write in BuildHuffmanTable](https://bugs.archlinux.org/task/79703). Other distributions might do something similar.

I don't see a way to disable WebP images in Chromium, unfortunately.

**Reply by The-Compiler** [reply]
Ah, I missed that Archlinux uses the system-wide libwebp install rather than the bundled one, and that's [already upgraded](https://gitlab.archlinux.org/archlinux/packaging/packages/libwebp/-/commit/22ec89407fb824622753ba6071ef6b8983580295).

I suppose other distributions do the same, as QtWebEngine defaults to using the system-wide libwebp if it's available anyways. [Debian does](https://packages.debian.org/bookworm/libqt6webenginecore6) for example, [Void Linux too](https://github.com/void-linux/void-packages/blob/master/srcpkgs/qt6-webengine/template), [Gentoo as well](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-qt/qtwebengine/qtwebengine-6.5.2-r1.ebuild).

So this should only really be a problem for Windows/macOS releases which have it bundled. I unfortunately lack the resources to build things from source there, but I plan to release v3.0.1 very soon after Qt 6.5.3 is out (as that fixes some other rather annoying bugs as well).

**Reply by trey-pindrop** [reply]
Any word on when the macOS package will be built?  I have an Intel MacBook Pro I can donate some time on if you need it, QtWebEngine 6.5.3 was released a couple of weeks ago.

**Reply by The-Compiler** [reply]
12 days, not a "couple of weeks". Hardware isn't a problem (releases are automated on GitHub Actions), but PyQt 6.5.3 (which is what the binary release is based on) [isn't out yet](https://www.riverbankcomputing.com/pipermail/pyqt/2023-October/045542.html). The release will follow as soon as that's the case, so hopefully later this week.

**Reply by The-Compiler** [reply]
Took me some more days after the [PyQt 6.5.3 release](https://www.riverbankcomputing.com/news/PyQt_v6.5.3_Released), but qutebrowser v3.0.2 based on it [is now released](https://github.com/qutebrowser/qutebrowser/releases/tag/v3.0.2).

**Comment by trey-pindrop** [comment]
Cool, thanks for the update.
